filters/shedder/admission.go

```diff
@@ -433,9 +433,12 @@ func (ac *admissionControl) Response(ctx filters.FilterContext) {
 		return
 	}
 
-	code := ctx.Response().StatusCode
-	if code < 499 {
-		ac.successCounter.Add(1)
+	rsp := ctx.Response()
+	if rsp != nil {
+		code := rsp.StatusCode
+		if code < 499 {
+			ac.successCounter.Add(1)
+		}
```

Comment:
I think this would not work as we translate proxyError caused by client into 499 response in the p.errorResponse

Reply:
Right now it works because rsp is nil , but requires change if I change to create a fakeResponse as you suggest in another comment.